### PR TITLE
PHP-16 

### PR DESCRIPTION
**Pull request by mpenick**

### CODE REVIEW

File: ext/src/Cassandra/Exception/DivideByZeroException.c [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+static zend_function_entry DivideByZeroException_methods[] = {
+  PHP_FE_END
+};
+
+void cassandra_define_DivideByZeroException(TSRMLS_D)
+{
+  zend_class_entry ce;
+
+  INIT_CLASS_ENTRY(ce, "Cassandra\\Exception\\DivideByZeroException", DivideByZeroException_methods);
+  cassandra_divide_by_zero_exception_ce = zend_register_internal_class_ex(&ce, cassandra_configuration_exception_ce, NULL TSRMLS_CC);
```
Comment (avalanche123) [inline ext/src/Cassandra/Exception/DivideByZeroException.c]: do you mind if we make this a subclass of `Cassandra\RangeException`?

File: ext/src/Cassandra/Types/Float.c [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  RETURN_DOUBLE((double) self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::minValue() */
+PHP_METHOD(Float, minValue)
```
Comment (avalanche123) [inline ext/src/Cassandra/Types/Float.c]: I like these methods, I've seed similar ones in other classes, what do you think about making the static? E.g.:

```php
<?php
$maxFloat = Cassandra\Float::max();
```

File: ext/src/Cassandra/Types/Float.c [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+{
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* flt =
+          (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  flt->value = LONG_LONG_MIN;
```
Comment (avalanche123) [inline ext/src/Cassandra/Types/Float.c]: Looks like this one is using a long instead of double

File: ext/src/Cassandra/Types/Float.c (hunk 2 above)
Comment (avalanche123) [inline ext/src/Cassandra/Types/Float.c]: Heh, never mind it is static

File: ext/src/Cassandra/Types/Float.c [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+ZEND_END_ARG_INFO()
+
+static zend_function_entry cassandra_float_methods[] = {
+  PHP_ME(Float, __construct, arginfo__construct, ZEND_ACC_CTOR|ZEND_ACC_PUBLIC)
+  PHP_ME(Float, __toString, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, value, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, isInfinite, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, isFinite, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, isNaN, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, add, arginfo_num, ZEND_ACC_PUBLIC)
```
Comment (avalanche123) [inline ext/src/Cassandra/Types/Float.c]: would be cool to typehint these methods to accept `Cassandra\Numeric` instances

File: ext/src/Cassandra/Types/Float.c (hunk 3 above)
Comment (mpenick) [inline ext/src/Cassandra/Types/Float.c]: woah, what's going on there. fixing.

File: ext/src/Cassandra/Types/Float.c (hunk 4 above)
Comment (mpenick) [inline ext/src/Cassandra/Types/Float.c]: I thought about that, but I was thinking these could also accept php long/double/string?

File: ext/src/Cassandra/Exception/DivideByZeroException.c (hunk 1 above)
Comment (mpenick) [inline ext/src/Cassandra/Exception/DivideByZeroException.c]: Makes sense. Changed.

File: ext/src/Cassandra/Types/Float.c (hunk 4 above)
Comment (avalanche123) [inline ext/src/Cassandra/Types/Float.c]: good point, we might relax the constraint later as that wouldn't be a BC break, but I think it makes sense to begin with a stricter interface

File: ext/src/Cassandra/Numeric.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  PHP_ABSTRACT_ME(Numeric, add, arginfo_num)
+  PHP_ABSTRACT_ME(Numeric, sub, arginfo_num)
+  PHP_ABSTRACT_ME(Numeric, mul, arginfo_num)
+  PHP_ABSTRACT_ME(Numeric, div, arginfo_num)
+  PHP_ABSTRACT_ME(Numeric, mod, arginfo_num)
+  PHP_ABSTRACT_ME(Numeric, abs, arginfo_none)
+  PHP_ABSTRACT_ME(Numeric, neg, arginfo_none)
+  PHP_ABSTRACT_ME(Numeric, sqrt, arginfo_none)
+  PHP_ABSTRACT_ME(Numeric, toInt, arginfo_none)
+  PHP_ABSTRACT_ME(Numeric, toDouble, arginfo_none)
```
Comment (avalanche123) [inline ext/src/Cassandra/Numeric.c:23]: This should be called `toFloat` to match PHP's type name. Is there a reason you didn't include `toVarint` `toDecimal` here?

Comment (mpenick) [inline ext/src/Cassandra/Numeric.c:23]: `toFloat()` is pretty confusing with there being a `Cassandra\Float` type. There are also "double" aliases for most "float" related things in PHP (unlike for "int"). It came down to time for `toVarint` etc. I didn't implement them so I didn't add them.

Comment (avalanche123) [inline ext/src/Cassandra/Numeric.c:23]: Got it, makes sense

**Pull request closed without merging**